### PR TITLE
v1/card: Add create card definition

### PR DESCRIPTION
**Pull request by nkristianto**
Add create card definition to accomodate create card operation

[OB-1930] #done

### CODE REVIEW

File: v1/card/card.gunk [hunk 1]
```diff
@@ -90,6 +87,16 @@ type Card struct {
 	AccessStatus types.CardAccessStatus `pb:"9" json:"access_status"`
 }
 
+// CardAttribute hold all detail about card attribute
+type CardAttribute struct {
+	// ID is a unique identifier of the attribute
```
Comment (jwangsadinata) [inline v1/card/card.gunk]: either remove this comment, or put comments on all of the other fields @nkristianto

File: v1/card/card.gunk [hunk 2]
```diff
@@ -126,8 +133,123 @@ type Result struct {
...
+	//
+	// +gunk http.Match{
+	//         Method: "POST",
+	//         Path:   "/v1/card/attribute",
+	//         Body:   "*",
+	// }
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"CardAttribute"},
+	//         Description: "Create new card attribute",
+	//         Summary:     "Create new card attribute ",
```
Comment (jwangsadinata) [inline v1/card/card.gunk]: remove the extra space after attribute.

File: v1/card/card.gunk [hunk 3]
```diff
@@ -165,11 +287,49 @@ type CardService interface {
...
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"Card"},
+	//         Description: "Retrieves all available cards for specific user, selected by the user_id",
+	//         Summary:     "Retrieves all available cards",
+	//         Responses: map[string]openapiv2.Response{
+	//                 "200": openapiv2.Response{
+	//                         Description: "Request executed successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/cardGetCardForSpecificUser",
```
Comment (jwangsadinata) [inline v1/card/card.gunk]: is it GetCardForSpecificUser or GetCardForCurrentUser?

**Review by jwangsadinata — APPROVED** [review]
some comments, but overall LGTM @nkristianto.

File: v1/card/card.gunk [hunk 4]
```diff
@@ -126,8 +133,123 @@ type Result struct {
...
+
+type DeleteCardRequest struct {
+	CardID string `pb:"1" json:"card_id"`
+}
+
+type GetCardForCurrentUserRequest struct {
+	UserID string `pb:"1" json:"user_id"`
+}
+
+type GetCardForCurrentUserResponse struct {
```
Comment (ghazninattarshah) [inline v1/card/card.gunk]: the results are slice, so this can be `cards` instead of `card`

File: v1/card/card.gunk (hunk 3 above)
Comment (ghazninattarshah) [reply v1/card/card.gunk]: You can define the function name as `GetUserCards` or `GetCardsByUserID`

File: v1/card/card.gunk [hunk 5]
```diff
@@ -165,11 +287,49 @@ type CardService interface {
...
+	//                         },
+	//                 },
+	//         },
+	// }
+	GetCardForCurrentUser(GetCardForCurrentUserRequest) GetCardForCurrentUserResponse
+
 	// UpdateCardStatus update the card status.
 	//
 	// +gunk http.Match{
 	//         Method: "POST",
```
Comment (ghazninattarshah) [inline v1/card/card.gunk]: use PUT for update operations. https://restfulapi.net/rest-put-vs-post/

**Review by ghazninattarshah — APPROVED** [review]
LGTM

File: v1/card/card.gunk (hunk 5 above)
Comment (nkristianto) [reply v1/card/card.gunk]: gotcha, thanks

**Pull request merged**